internal/rpmmd/repository.go

```diff
@@ -97,11 +97,13 @@ func (pkg Package) ToPackageInfo() PackageInfo {
 	}
 }
 
-// The inputs to depsolve, a set of packages to include and a set of
-// packages to exclude.
+// The inputs to depsolve, a set of packages to include and a set of packages
+// to exclude. The Repositories are used when depsolving this package set in
+// addition to the base repositories.
 type PackageSet struct {
-	Include []string
-	Exclude []string
+	Include      []string
+	Exclude      []string
+	Repositories []RepoConfig
```

Comment:
The thought did cross my mind, but I think it also makes sense as it is.  I don't think the use cases are that distinct.  A `PackageSet` is a list of packages to be included and excluded.  The concept of excluding packages only makes sense in the context of depsolving; the purpose of the class is to depsolved, and repository configurations are an integral part of that.

I can see your point to an extent, especially if we consider where package sets are born: in `distro/*/package_sets.go`.  The idea that those definitions can have repositories attached to them is weird, because they're just static package lists that have no knowledge of repositories.  That said, I think an extra type might just make noise for no real benefit, since the package sets returned from `imageType.PackageSets()` will immediately be of the _new_ type (that includes repositories).  The `PackageSet` type without repositories would only be used to define the package lists in the distros.

Reply:
Yeah, fair enough. We could also have a new type, which would be used in image definitions and have that one added to the `PackageSet` type and keep using `PackageSet` everywhere as we do now. Something in the sense of `BaseSolver` and `Solver` added by this PR.

But I agree that this is a bit unnecessary (does not bring much of a value) and over-engineering so I'm OK with keeping it as it is now 😉